dbt/adapters/bigquery/connections.py

```diff
@@ -558,89 +432,148 @@ def copy_bq_table(self, source, destination, write_disposition):
             write_disposition,
         )
 
-        def copy_and_results():
-            job_config = google.cloud.bigquery.CopyJobConfig(write_disposition=write_disposition)
-            copy_job = client.copy_table(source_ref_array, destination_ref, job_config=job_config)
-            timeout = self.get_job_execution_timeout_seconds(conn) or 300
-            iterator = copy_job.result(timeout=timeout)
-            return copy_job, iterator
-
-        self._retry_and_handle(
-            msg='copy table "{}" to "{}"'.format(
-                ", ".join(source_ref.path for source_ref in source_ref_array),
-                destination_ref.path,
-            ),
-            conn=conn,
-            fn=copy_and_results,
+        msg = 'copy table "{}" to "{}"'.format(
+            ", ".join(source_ref.path for source_ref in source_ref_array),
+            destination_ref.path,
+        )
+        with self.exception_handler(msg):
```

Comment:
this changes the exception handling so we should wrap that in the original retry predicate logic somehow